### ISSUE TITLE
[BUG][v2]: SumAggregation takes 4 positional arguments instead of 3 error when running train from CLI

### ISSUE THREAD

**Issue by joelnkn** [chemprop/chemprop#705]
Script ran:

```
results_dir=results_qm9_gap
train_path=../data/qm9/train.csv
val_path=../data/qm9/val.csv
test_path=../data/qm9/test.csv

#Hyperparameter optimization
# chemprop hyperopt \
# --dataset-type regression \
# --data-path $train_path \
# --separate-val-path $val_path \
# --separate-test-path $val_path \
# --num-iters 30 \
# --epochs 50 \
# --aggregation norm \
# --search-parameter-keywords depth ffn_num_layers  hidden_size ffn_hidden_size dropout \
# --config-save-path $results_dir/config.json \
# --hyperopt-checkpoint-dir $results_dir \
# --log-dir $results_dir \
# --target-columns gap

#Training with optimized hyperparameters
chemprop train \
-t regression \
--data-path $train_path \
--separate-val-path $val_path \
--separate-test-path $test_path \
--epochs 50 \
--aggregation norm \
--config-path $results_dir/config.json \
--save-dir $results_dir \
--ensemble-size 5 \
--save-preds \
--metrics mae \
--show-individual-scores \
--target-columns gap
```

Output:
```
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
Missing logger folder: results_qm9_gap/model_0/trainer_logs
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [GPU-374034c4-2c4f-3c8a-830c-085bf28b29a1]
Loading `train_dataloader` to estimate number of stepping batches.

  | Name            | Type               | Params
-------------------------------------------------------
0 | message_passing | BondMessagePassing | 264 K 
1 | agg             | NormAggregation    | 0     
2 | bn              | BatchNorm1d        | 600   
3 | predictor       | RegressionFFN      | 90.6 K
  | other params    | n/a                | 1     
-------------------------------------------------------
355 K     Trainable params
1         Non-trainable params
355 K     Total params
1.422     Total estimated model params size (MB)
SLURM auto-requeueing enabled. Setting signal handlers.

  File "/home/gridsan/jmanu/.conda/envs/chemprop/bin/chemprop", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/main.py", line 69, in main
    func(args)
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/train.py", line 50, in func
    main(args)
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/train.py", line 807, in main
    train_model(args, train_loader, val_loader, test_loader, output_dir, scaler)
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/train.py", line 732, in train_model
    trainer.fit(model, train_loader, val_loader)
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/trainer/trainer.py", line 543, in fit
    call._call_and_handle_interrupt(
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/trainer/call.py", line 44, in _call_and_handle_i
nterrupt
    return trainer_fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/trainer/trainer.py", line 579, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/trainer/trainer.py", line 986, in _run
    results = self._run_stage()
              ^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/trainer/trainer.py", line 1030, in _run_stage
    self._run_sanity_check()
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/trainer/trainer.py", line 1059, in _run_sanity_c
heck
    val_loop.run()
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/loops/utilities.py", line 182, in _decorator
    return loop_run(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 135, in run
    self._evaluation_step(batch, batch_idx, dataloader_idx, dataloader_iter)
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 396, in _evaluat
ion_step
    output = call._call_strategy_hook(trainer, hook_name, *step_args)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/trainer/call.py", line 309, in _call_strategy_ho
ok
    output = fn(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/strategies/strategy.py", line 412, in validation
_step
    return self.lightning_module.validation_step(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/models/model.py", line 157, in validation_step
    losses = self._evaluate_batch(batch)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/models/model.py", line 174, in _evaluate_batch
    preds = self(bmg, V_d, X_f)
            ^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/models/model.py", line 140, in forward
    return self.predictor(self.fingerprint(bmg, V_d, X_f))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/models/model.py", line 125, in fingerprint
    H = self.agg(H_v, bmg.batch)
        ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/gridsan/jmanu/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/nn/agg.py", line 105, in forward
    return super().forward(H, batch, self.dim) / self.norm
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: SumAggregation.forward() takes 3 positional arguments but 4 were given
```

**Comment by hwpang** [comment]
Thanks for catching this! I will look into this.

**Comment by kevingreenman** [comment]
fixed by #707
